src/main/java/org/jabref/logic/importer/fileformat/BibtexParser.java

```diff
@@ -386,7 +390,23 @@ private void parseJabRefComment(Map<String, String> meta) {
             } catch (ParseException ex) {
                 parserResult.addException(ex);
             }
+        } else if (comment.substring(0, Math.min(comment.length(), MetaData.META_FLAG_VERSION_010.length())).equals(MetaData.META_FLAG_VERSION_010)) {
+            parseCommentToJson(comment, meta);
+        }
+    }
+
+    private JsonObject parseCommentToJson(String comment, Map<String, String> meta) {
```

Comment:
Remove the map as parameter. It should not be modified.

The JsonObject should be separate - to enable handling v5.x meta data (Map) and v6.x metadata (JSON)

Reply:
Moreover, return `Optional<JsonObject>` (which is more modern Java)